internal/ls/completions.go

```diff
@@ -3023,22 +3003,30 @@ func getContextualType(previousToken *ast.Node, position int, file *ast.SourceFi
 		if ast.IsConditionalExpression(parent) {
 			return getContextualTypeForConditionalExpression(parent, position, file, typeChecker)
 		}
-		// Fall through to default for other colon contexts (object literals, etc.)
-		fallthrough
-	default:
-		argInfo := getArgumentInfoForCompletions(previousToken, position, file, typeChecker)
-		if argInfo != nil {
-			return typeChecker.GetContextualTypeForArgumentAtIndex(argInfo.invocation, argInfo.argumentIndex)
-		} else if isEqualityOperatorKind(previousToken.Kind) && ast.IsBinaryExpression(parent) && isEqualityOperatorKind(parent.AsBinaryExpression().OperatorToken.Kind) {
-			// completion at `x ===/**/`
-			return typeChecker.GetTypeAtLocation(parent.AsBinaryExpression().Left)
-		} else {
-			contextualType := typeChecker.GetContextualType(previousToken, checker.ContextFlagsCompletions)
-			if contextualType != nil {
-				return contextualType
+	case ast.KindCommaToken:
```

Comment:
I might still co-locate the `[` and `]` cases, but this is fine.

I was thinking about whether there are cases where we should also guard on `]`, but I couldn't imagine where you'd want or have a contextual type immediately after a `]`.